doc/sphinx/proof-engine/ltac.rst

```diff
@@ -1661,6 +1661,100 @@ Testing boolean expressions: guard
    .. exn:: Condition not satisfied.
       :undocumented:
 
+Checking properties of terms
+----------------------------
+
+Each of the following tactics acts as the identity if the check
+succeeds, and results in an error otherwise.
+
+.. tacn:: constr_eq @one_term @one_term
+
+   Checks whether its arguments are equal ignoring differences in alpha
+   conversion, casts and universe constraints. It may unify universes.
+
+   .. exn:: Not equal.
+      :undocumented:
+
+   .. exn:: Not equal (due to universes).
+      :undocumented:
+
+.. tacn:: constr_eq_strict @one_term @one_term
+
+   This tactic checks whether its arguments are equal modulo alpha
+   conversion, casts and universe constraints. It does not add new
+   constraints.
+
+   .. exn:: Not equal.
+      :undocumented:
+
+   .. exn:: Not equal (due to universes).
+      :undocumented:
+
+.. tacn:: constr_eq_nounivs @one_term @one_term
+
+   Checks whether its arguments are equal modulo alpha
+   conversion, casts and universe constraints. It does not add new
+   constraints.
+
+.. tacn:: unify @one_term @one_term {? with @ident }
+
+   Checks whether its arguments are unifiable, potentially
+   instantiating existential variables.
+
+   Unification takes the transparency information defined in the hint database
+   :n:`@ident` into account (see :ref:`the hints databases for auto and eauto
+   <thehintsdatabasesforautoandeauto>`).
+
+.. tacn:: is_evar @term
+   :name: is_evar
+
+   Checks whether its argument is a current existential
+   variable. Existential variables are uninstantiated variables generated
+   by :tacn:`eapply` and some other tactics.
+
+   .. exn:: Not an evar.
+      :undocumented:
+
+.. tacn:: has_evar @term
+   :name: has_evar
+
+   Checks whether its argument has an existential variable as
+   a subterm. Unlike context patterns combined with ``is_evar``, this tactic
+   scans all subterms, including those under binders.
+
+   .. exn:: No evars.
+      :undocumented:
+
+.. tacn:: is_var @term
+   :name: is_var
+
+   Checks whether its argument is a variable or hypothesis in
+   the current local context.
+
+   .. exn:: Not a variable or hypothesis.
+      :undocumented:
+
+.. tacn:: is_cofix @one_term
+   :undocumented:
+
+.. tacn:: is_const @one_term
+   :undocumented:
+
+.. tacn:: is_constructor @one_term
+   :undocumented:
+
+.. tacn:: is_fix @one_term
+   :undocumented:
+
+.. tacn:: is_ground @one_term
+   :undocumented:
+
+.. tacn:: is_ind @one_term
+   :undocumented:
```

Comment:
```suggestion
   :name: is_ind
   
   Checks whether its argument is a (co)inductive type (family).
   
   .. exn:: not an (co)inductive datatype.
      :undocumented:
```

Here's an example, if you want to work it in:
```coq
Goal True.
  is_ind prod.
```
Note that `is_ind (list nat)`, for example, fails.

Reply:
`is_ind list` succeeds, so it's odd that it fails for `list nat`, which I would think is an inductive type.  Is this correct behavior for `list nat`?  Is there an understandable explanation (though I could just note that it doesn't work)?

Reply:
Yes, the behavior is correct.  All of these tactics are about the top-most / root node of the AST.  The root node of `list nat` is an application, not an inductive.